daemon/k8s_watcher.go

```diff
@@ -408,9 +409,19 @@ func (d *Daemon) serviceAddFn(obj interface{}) {
 		return
 	}
 
+	log.Debugf("Adding k8s service %+v", svc)
+
+	nodePortSvcs := types.K8sServiceList{}
+
 	switch svc.Spec.Type {
-	case v1.ServiceTypeClusterIP, v1.ServiceTypeNodePort, v1.ServiceTypeLoadBalancer:
-		break
+	// LoadBalancer and NodePort make the service available on the nodePort
+	// on each node. This creates a service with both the IPv4 and IPv6
+	// address of the host as the frontend.
```

Comment:
The commentary throughout this patch is very clear; this is a model for quality for what should be done throughout the project!